formats/sarifutils/sarifutils.go

```diff
@@ -234,6 +234,21 @@ func GetRuleFullDescription(rule *sarif.ReportingDescriptor) string {
 	return ""
 }
 
+func GetRuleProperty(key string, rule *sarif.ReportingDescriptor) string {
+	if rule != nil && rule.Properties != nil && rule.Properties[key] != nil {
+		prop, ok := rule.Properties[key].(string)
+		if !ok {
+			return ""
+		}
+		return prop
+	}
+	return ""
+}
+
+func GetRuleUndeterminedReason(rule *sarif.ReportingDescriptor) string {
+	return GetRuleProperty("undetermined_reason", rule)
```

Comment:
The function split is positive overall - but I don't see any other usage for `GetRuleProperty`, so in that case, I don't see a reason for the dedicated fucntion.
Couldn't we use it for the applicability status itself?

Reply:
We can use it for applicability status as it is also received in the Rule property. Do you want me to change applicability fetch to this function ?

Reply:
yes - otherwise the usage of the this generic function just one misses the point IMO

Reply:
Yes please do use the new `GetRuleProperty` when applicable. but the `GetRuleUndeterminedReason` does not belong here. please move it to `jasutils` package.
This files is only for utils to access, create or set values in sarif